chain/x/zoracle/exec_env.go

```diff
@@ -65,19 +111,25 @@ func (env *ExecutionEnvironment) GetValidatorAddress(validatorIndex int64) ([]by
 }
 
 func (env *ExecutionEnvironment) GetMaximumResultSize() int64 {
-	return env.keeper.MaxResultSize(env.ctx)
+	return env.maxResultSize
 }
 
 func (env *ExecutionEnvironment) GetMaximumCalldataOfDataSourceSize() int64 {
-	return env.keeper.MaxCalldataSize(env.ctx)
+	return env.maxCalldataSize
 }
 
 func (env *ExecutionEnvironment) RequestExternalData(
 	dataSourceID int64,
 	externalDataID int64,
 	calldata []byte,
-) error {
-	return env.keeper.AddNewRawDataRequest(env.ctx, env.requestID, types.ExternalID(externalDataID), types.DataSourceID(dataSourceID), calldata)
+) {
+	if int64(len(calldata)) <= env.maxCalldataSize {
+		env.rawDataRequests = append(env.rawDataRequests, types.NewRawDataRequestWithExternalID(
```

Comment:
Let make this a no-op when the size of `rawDataRequests` already exceeds max raw data request count, so Wasm cannot flood us with shit tons of raw requests.